rebaser.go

```diff
@@ -137,6 +142,71 @@ func (r *Rebaser) validateRebaseable(appImg imgutil.Image) error {
 		if !r.Force && rebaseable == "false" {
 			return fmt.Errorf("app image is not marked as rebaseable")
 		}
+
+		// check the OS, architecture, and variant values
+		// if they are not the same, the image cannot be rebased unless the force flag is set
+		if !r.Force {
+			appOS, err := appImg.OS()
+			if err != nil {
+				return errors.Wrap(err, "get app image os")
+			}
+			newBaseOS, err := newBaseImg.OS()
+			if err != nil {
+				return errors.Wrap(err, "get new base image os")
+			}
+			if appOS != newBaseOS {
+				return fmt.Errorf("incompatible os: '%s' is not compatible with '%s'", newBaseOS, appOS)
```

Comment:
nit here, and i think also applicable a few other places (even arch)  we're doing an equality check but the word we're using is "compatible" -- so i could be an ubuntu-based distro that promises 100% compatibility, but if i list my os as "mint" or whatever then the string won't equal ubuntu ...   and it's a nit b/c I'm not arguing we should be maintaining a detailed compatiblity table in the code, but I do think we should change the language to `is not equal to` so that it's more obvious to anyone confronting this error message that we're just doing the simplest thing

Reply:
Updated error message in f4092ea5